lib/services/api/request_parser.rb

```diff
@@ -10,7 +10,7 @@ def self.parse_user(data)
 
     def self.parse_options(data)
       raise BadRequestError, "Request is missing options" if data["options"].blank?
-      data["options"].symbolize_keys
+      data["options"].deep_symbolize_keys
```

Comment:
Can API save the data in the same format as what UI does? Currently the data saved via UI is like this.
```
---
:src_ids:
- '7000000000029'
:vm_memory: 1024
:disk_add:
- !ruby/hash:ActiveSupport::HashWithIndifferentAccess
  disk_size_in_mb: '1'
  persistent: true
  thin_provisioned: true
  new_controller_type: VirtualLsiLogicController
  dependent: true
  bootable: false
:request_type: :vm_reconfigure
```

Reply:
@lfu we *could* possibly convert to yaml...however, at the moment it is not playing nice:
```
TypeError: no implicit conversion of Symbol into Integer
from /Users/jilliantullo/code/manageiq/app/models/mixins/miq_request_mixin.rb:4:in `[]'
```
After looking at it, it appears that the `create_request` method we are [calling](https://github.com/ManageIQ/manageiq-api/blob/aca3a26078bfe0875327bc989561d34a109f649f/app/controllers/api/requests_controller.rb#L23) is expecting a hash.

Reply:
@jntullo ```options``` is a serialized column. So DB saves the ```options``` data in yaml format. I was hoping to keep the data format in DB the same as what UI does. ```Hash``` would work if the format can't be exactly the same. Currently all the callers use the options key as a symbol.